File: zad2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

void QueuePeoples(vector<pair<int, int>>& visitors, int numWindows) { // ф-ия формирования очереди
    vector<int> WindowsTimes(numWindows, 0);
    vector<int> WindowsNumbers(numWindows);

    for(int i= 0; i < numWindows; ++i) { // работа с номерами окон
        WindowsNumbers[i] = i + 1;
    }

    for (const auto& people : visitors) {
        
        auto MinWindowIndex = min_element(WindowsTimes.begin(), WindowsTimes.end()) - WindowsTimes.begin(); // находим индекс окна с минимальным временем работы

        cout << ">>> " << "Окно " << WindowsNumbers[MinWindowIndex] << " (" << WindowsTimes[MinWindowIndex] << " минут): "; // вывод информации по очереди
        cout << "T" << people.second;
        WindowsTimes[MinWindowIndex] += people.first; // Увеличиваем время работы окна на время обслуживания посетителя
        cout << endl;
    }
}

int main(){
    setlocale(LC_ALL, "Rus");

    vector<pair<int, int>> Peoples; // вектор пара, в котором будут хранится время и талоны посетителей
    cout << ">>> Введите кол-во рабочих окон:" << endl;
    cout << "<<< ";
    int NumsWindows;
    cin >> NumsWindows;

    cout << ">>> Для получения талона введите: ENQUEUE и время посещения\n\tДля формирования очереди введите: DISTRIBUTE\n\t\tДля выхода введите: EXIT" << endl;
    string command;
    bool k = true; // условие выхода из while
    int time; // время посетителя
    int talon = 100; // талон посетителя
    while (k == true) {
        cout << "<<< ";
        cin >> command;
        if (command == "ENQUEUE") {
            cin >> time;
            talon += 10;
            cout << ">>> T" << talon << endl;
            Peoples.push_back({time, talon});
        } else if (command == "DISTRIBUTE") {
            QueuePeoples(Peoples, NumsWindows); // ф-ия формирования очереди
        } else if (command == "EXIT") {
            cout << ">>> Выход..." << endl;
            k = false;
        } else {
            cout << ">>> Команда введена неверно" << endl;
        }
    }

    return 0;
}

File: zad2.py
def queue_peoples(visitors, num_windows):
    windows_times = [0] * num_windows
    windows_numbers = list(range(1, num_windows + 1))

    for people in visitors:
        min_window_index = windows_times.index(min(windows_times))
        print(f">>> Окно {windows_numbers[min_window_index]} ({windows_times[min_window_index]} минут): T{people[1]}")
        windows_times[min_window_index] += people[0]
        

peoples = []
print(">>> Введите кол-во рабочих окон:")
num_windows = int(input("<<< "))

print(">>> Для получения талона введите: ENQUEUE и время посещения\n\tДля формирования очереди введите: DISTRIBUTE\n\t\tДля выхода введите: EXIT")
k = True
talon = 100
while k:
    command = input("<<< ")
    if command == "ENQUEUE":
        time = int(input())
        talon += 10
        print(f">>> T{talon}")
        peoples.append((time, talon))
    elif command == "DISTRIBUTE":
        queue_peoples(peoples, num_windows)
    elif command == "EXIT":
        print(">>> Выход...")
        k = False
    else:
        print(">>> Команда введена неверно")